bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java

```diff
@@ -138,15 +156,39 @@ public void removeStep(Long projectId, Long suitId, Long caseId, Long stepId) {
 
         Step step = stepDAO.findById(stepId).orElseThrow(NotFoundException::new);
         throwExceptionIfStepIsNotInCase(caze, step);
+
+        StepSuggestion stepSuggestion = stepSuggestionService.getStepSuggestion(projectId, step);
+
+        stepSuggestion.remove(step);
         caze.removeStep(step);
         suit.updateStatus();
 
+        projectStepSuggestionDAO.save(stepSuggestion);
         stepDAO.delete(step);
-
         caseVersionDAO.save(caze);
         suitVersionDAO.save(suit);
     }
 
+    public void removeStep(Long projectId, Long stepId) {
+        Project project = checkNotNull(projectService.getProjectByProjectId(projectId));
+
+        for (Suit suit: project.getSuits()) {
+            for (Case caze: suit.getCases()) {
+                for (Step step: caze.getSteps()) {
+                    if(stepId.equals(step.getId())){
+                        caze.removeStep(step);
```

Comment:
What about ConcurrentModificationException? Can you check it? https://stackoverflow.com/questions/10431981/remove-elements-from-collection-while-iterating/10432084

Reply:
There is return statement after removing a step, so iterator won't take new element after collection modification and ConcurrentModificationException won't be thrown